video/video_decoder.cpp

```diff
@@ -112,6 +112,15 @@ bool VideoDecoder::needsUpdate() const {
 	return false;
 }
 
+void VideoDecoder::delayMillis(uint msecs) {
+	if (!needsUpdate())
+		g_system->delayMillis(MIN<uint>(msecs, getTimeToNextFrame()));
+#ifdef __DS__
```

Comment:
`ATARI` would benefit from this as well, I'm thinking whether this shouldn't be specified in a more generic way.